### PR TITLE
Add HookParameters fee calc to all txs

### PR DESCRIPTION
**Pull request by dangell7**
### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/ripple/app/tx/impl/Transactor.cpp [hunk 1]
```diff
@@ -350,6 +350,23 @@ Transactor::calculateBaseFee(ReadView const& view, STTx const& tx)
...
+        for (auto const& param : params)
+        {
+            paramBytes += (param.isFieldPresent(sfHookParameterName)
+                               ? param.getFieldVL(sfHookParameterName).size()
+                               : 0) +
+                (param.isFieldPresent(sfHookParameterValue)
+                     ? param.getFieldVL(sfHookParameterValue).size()
+                     : 0);
+        }
+        accumulator += XRPAmount{static_cast<XRPAmount>(paramBytes)};
```
Comment (RichardAH) [inline src/ripple/app/tx/impl/Transactor.cpp:367]: these used to have overflow checks didn't they? what happened to that logic?

**Review by RichardAH — CHANGES_REQUESTED** [review]

File: src/ripple/app/tx/impl/Transactor.cpp (hunk 1 above)
Comment (dangell7) [reply src/ripple/app/tx/impl/Transactor.cpp:367]: The overflow checks were in SetHook.cpp but not in Invoke.cpp. I can add them here though

File: src/ripple/app/tx/impl/SetHook.cpp [hunk 2]
```diff
@@ -566,40 +566,6 @@ SetHook::calculateBaseFee(ReadView const& view, STTx const& tx)
             createFee = XRPAmount{hook::computeCreationFee(
                 hookSetObj.getFieldVL(sfCreateCode).size())};
 
-        XRPAmount paramFee{0};
```
Comment (RichardAH) [inline src/ripple/app/tx/impl/SetHook.cpp]: this code is adding up the cost to submit the txn based on the hook parameters inside the hooks array inside hookset, this is a different bill and different parameters to the top level hook parameters array so it should be left in

Comment (dangell7) [reply src/ripple/app/tx/impl/SetHook.cpp]: Update, thank you!

**Review by RichardAH — APPROVED** [review]

**Pull request merged**